sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridEventConverter.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core;
+using Azure.Messaging.EventGrid.Models;
+
+namespace Azure.Messaging.EventGrid
+{
+    /// <summary>
+    /// A custom converter that attributes the <see cref="EventGridEvent"/> type.
+    /// This allows System.Text.Json to serialize and deserialize EventGridEvent by default.
+    /// </summary>
+    internal class EventGridEventConverter : JsonConverter<EventGridEvent>
```

Comment:
This is tested by existing tests after switching the Parse and Publish methods to use this.